jenkins/jobs/jenkins_pipeline_sample.groovy

```diff
@@ -43,11 +44,24 @@ String mySqlRootCredential = binding.variables["MYSQL_ROOT_CREDENTIAL_ID"] ?: ""
 String mySqlCredential = binding.variables["MYSQL_CREDENTIAL_ID"] ?: ""
 // remove::end[K8S]
 
+ScmContext.metaClass.configureScm = { repoId, branchId ->
```

Comment:
Hmm I think we can try to rewrite it in order no to use meta programming. WDYT?

Reply:
Sth like (I don't really remember the correct classes)

```groovy
Closure configureScm = { DslContext context, String repoId, String branchId ->  
context.git {
		remote {
			name('origin')
			url(repoId)
			branch(branchId)
			credentials(gitUseSshKey ? gitSshCredentials : gitCredentials)
		}
		extensions {
			wipeOutWorkspace()
		}
	}
}
```

then you would call it like this

```groovy
configureScm(delegate as DslContext, fullGitRepo, branchName)
```

Reply:
We could definitely do it if you have reservations about metaprogramming. The reason I like it in this scenario is that it basically uses the same mechanism as the Jenkins DSL itself thus making it look more natural. It also guarantees that we could only call the closure from the proper DSL block only and the error messages are more natural.

What are your concerns about adding the closure directly to the ScmContext?